tests/python/pants_test/backend/python/tasks/test_python_distribution_integration.py

```diff
@@ -53,9 +56,25 @@ def test_pants_run(self):
       # Check that text was properly printed to stdout.
       self._assert_native_greeting(pants_run.stdout_data)
 
+  @staticmethod
+  def rewrite_hello_c_source(orig_content):
+    return re.sub('"Hello from C!"', '"Hello from C?"', orig_content)
+
+  def test_invalidation(self):
+    run_command = ['run', '{}:main_with_no_conflict'.format(self.fasthello_install_requires)]
+    unmodified_pants_run = self.run_pants(command=run_command)
+    self.assert_success(unmodified_pants_run)
+    self.assertIn('Hello from C!\n', unmodified_pants_run.stdout_data)
+
+    c_source_file = '{}/c_greet.c'.format(self.fasthello_install_requires)
+    with self.rewrite_file_content(c_source_file, self.rewrite_hello_c_source):
```

Comment:
non-blocking: since you are modifying the content of the repo, it might create surprises if later tests happen to run against the touched content. It's best to create a target on the test run, but if it's hard to do, it would be good to restore it.

Reply:
Yes, and I've had the file stuck in the intermediate "rewritten" state at least once while testing -- I'm not sure what caused that since it is wrapped in a `try` but I'll see if I can change the implementation to make a target for the test (and make sure that that implementation fails on current master so it's still a valid test). I have a hunch this won't be that big of a change, and for the reasons you mentioned I would prefer not to leave the `rewrite_file_content` method in the integration test base class.

Reply:
Ended up using `mock_buildroot()` from `PantsRunIntegrationTest`, which is much more clear anyway -- see the new test [here](https://github.com/cosmicexplorer/pants/blob/83429ecd725ef49f195a5e5bf6f9ee195f340049/tests/python/pants_test/backend/python/tasks/test_python_distribution_integration.py#L60). Thanks for pointing this out!